### PR TITLE
Share: Add invite user to the sharing menu

### PR DESCRIPTION
**Pull request by evictorero**
**What is this feature?**

Adds the option to invite user to the sharing menu when the `externalUserMngLinkUrl` is defined and the user has `OrgUsersAdd` permissions.
It uses a similar condition that we have in the CTA `Manage users on grafana.com` in the `admin/users` page.
The only difference is a user with `OrgUsersRead` can `manage users on grafana.com` but in this case, it will only accessible to those who has `OrgUsersAdd`.

image

Rudderstack event when clicked:
```
interactionName: "dashboards_sharing_category_clicked"
properties: {item: "invite_user", shareResource: "dashboard"}
```

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana-enterprise/issues/7603

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by natydej** [comment]
I think that the option should be re-named as **"Invite new member"**, to clarify the purpose, also because in the `grafana.com` page, it will redirect to this form below:
Grafana comform

Another quick thing here, if this link is going to be open in a new page, I would recommend to use the `"external-link-alt"` icon, that will communicate users that they will leave the current page.

One last suggestion, regarding the order of the dropdown menu list, I think we should consider to be something like:

1. Share internally
2. Share externally
3. Share snapshot
4. Schedule report
5. Invite new member

**Why?**

- Where all sharing options are the most frequent and closely related actions and by keeping them together maintains a better user flow. 
- Inviting new users, is conceptually different (organizational rather than dashboard-specific), so placing it at the bottom separates it from sharing tasks.

Here is an image with the idea below:

Screenshot 2024-11-27 at 09 41 59

**Comment by evictorero** [comment]
Thanks @natydej for the feedback, all changes were applied.
I also like the new order menu because it keeps together, in the beginning, all OSS options.

**Review by natydej — APPROVED** [review]
LGTM!

**Comment by juanicabanas** [comment]
> enu list, I think

@natydej what about adding a divider between dashboard-scope actions and this new button?
image

File: public/app/features/dashboard-scene/sharing/ShareButton/ShareMenu.tsx [hunk 1]
```diff
@@ -79,6 +77,20 @@ export default function ShareMenu({ dashboard, panel }: { dashboard: DashboardSc
...
 
+    customShareDrawerItems.forEach((d) => menuItems.push(d));
+
+    menuItems.push({
+      shareId: shareDashboardType.inviteUser,
+      testId: newShareButtonSelector.inviteUser,
+      icon: 'external-link-alt',
+      label: t('share-dashboard.menu.invite-user-title', 'Invite new member'),
+      renderCondition:
+        config.externalUserMngLinkUrl !== '' && contextSrv.hasPermission(AccessControlAction.OrgUsersAdd),
```
Comment (juanicabanas) [inline public/app/features/dashboard-scene/sharing/ShareButton/ShareMenu.tsx:88]:
```suggestion
        !!config.externalUserMngLinkUrl && contextSrv.hasPermission(AccessControlAction.OrgUsersAdd),
```
you check the truthiness. This also handles cases like null and undefined, making it more robust

**Review by juanicabanas — APPROVED** [review]
Tested and works as expected.
LGTM!
I left a small nit

**Comment by natydej** [comment]
@juanicabanas about your comment, yes, that's another option we could explore. But instead of being at the bottom, it could be at the top, to make it more prominent and for users more discoverable as a new option.
This is how it will look like:

**Option A**
Having the icon as a blue color to represent is a link, and keep the label as primary color
Screenshot 2024-11-28 at 14 42 10

**Option B**
Having the icon and the label as link blue color
Screenshot 2024-11-28 at 14 43 16

**Comment by natydej** [comment]
Considering another design solution where we all thinking about scalability of this item. Putting in at the bottom makes more sense since is a new option added in the dropdown menu, and also it will be available for only admin users. About the icons, we can use the `"add-user"` icon to represent the action itself of **"Invite new member"** and also adding at the end the `"external-link-alt"` icon to represent will be open in a new tab now, but in the future we could remove it since it will be open in a drawer as other actions in that menu.
Screenshot 2024-12-17 at 11 31 59

File: public/app/features/dashboard-scene/sharing/ShareButton/ShareMenu.tsx [hunk 2]
```diff
@@ -109,16 +116,32 @@ export default function ShareMenu({ dashboard, panel }: { dashboard: DashboardSc
...
-          {item.renderBottomDivider && <Menu.Divider />}
         </React.Fragment>
       ))}
     </Menu>
   );
 }
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
+    item: css({
```
Comment (juanicabanas) [inline public/app/features/dashboard-scene/sharing/ShareButton/ShareMenu.tsx:137]: rename to `inviteUserItem`

File: public/app/features/dashboard-scene/sharing/ShareButton/ShareMenu.tsx [hunk 3]
```diff
@@ -109,16 +116,32 @@ export default function ShareMenu({ dashboard, panel }: { dashboard: DashboardSc
...
+      display: 'flex',
+      justifyContent: 'start',
+      flexDirection: 'row',
+      alignItems: 'center',
+    }),
+    icon: css({
```
Comment (juanicabanas) [inline public/app/features/dashboard-scene/sharing/ShareButton/ShareMenu.tsx:143]: `inviteUserIcon` ?

**Review by juanicabanas — APPROVED** [review]

**Pull request merged**